### PR TITLE
Use local variable in btree2 and print value

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

File: test/btree2.c [hunk 1]
```diff
@@ -9917,19 +9917,21 @@ main(void)
...
     /* Reset library */
     h5_test_init();
-    fapl = h5_fileaccess();
+    fapl             = h5_fileaccess();
+    localTestExpress = TestExpress;
 
     /* For the Direct I/O driver, skip intensive tests due to poor performance */
-    if (!strcmp(driver_name, "direct"))
-        SetTestExpress(2);
+    if (localTestExpress < 2 && strcmp(driver_name, "direct"))
```
Comment (fortnern) [inline test/btree2.c]: As Larry mentioned in the meeting I think this should be !strcmp(...

File: test/btree2.c [hunk 2]
```diff
@@ -9917,19 +9917,21 @@ main(void)
...
+        localTestExpress = 2;
 
-    if (TestExpress > 1)
-        printf("***Express test mode on.  Some tests may be skipped\n");
+    if (localTestExpress > 0)
```
Comment (fortnern) [inline test/btree2.c:9933]: Why the change from 1 to 0 here? All the other places in this PR use 1.

File: test/earray.c [hunk 3]
```diff
@@ -2303,7 +2303,7 @@ main(void)
     h5_test_init();
     fapl = h5_fileaccess();
     if (TestExpress > 1)
-        printf("***Express test mode on.  Some tests may be skipped\n");
+        printf("***Express test mode %d.  Some tests may be skipped\n", TestExpress);
```
Comment (fortnern) [inline test/earray.c:2306]: I prefer to explicitly specify the type of printf value parameters (int here) because there is absolutely no type checking, and a change in the global variable type could cause hard to track down errors. The LocalTestExpress ones aren't as bad since that variable is declared in the function but it wouldn't hurt to add it there too.

File: test/btree2.c (hunk 1 above)
Comment (byrnHDF) [reply test/btree2.c]: Fixed

File: test/btree2.c (hunk 2 above)
Comment (byrnHDF) [reply test/btree2.c:9933]: Technically, level one could skip tests. The other uses actually skip tests if level 2 or 3.

File: test/earray.c (hunk 3 above)
Comment (byrnHDF) [reply test/earray.c:2306]: Not sure what you want here, since the variable is just an int?

File: test/btree2.c (hunk 2 above)
Comment (byrnHDF) [reply test/btree2.c:9933]: > Technically, level one could skip tests. The other uses actually skip tests if level 2 or 3.

Fixed other files.

File: test/earray.c (hunk 3 above)
Comment (qkoziol) [reply test/earray.c:2306]: Cast LocalTestExpress to an int:  "LocalTestExpress" => "(int)LocalTestExpress"

Comment (jhendersonHDF) [reply test/earray.c:2306]: Seems like a bad idea to cast something that doesn't really need casting since it'll just hide future possible errors.

Comment (fortnern) [reply test/earray.c:2306]: If the type of the variable ever changes it can produce memory errors. Modern compilers might be able to detect and throw a warning in this case (if the type doesn't match the format string), I'm not sure.

Comment (byrnHDF) [reply test/earray.c:2306]: I think you want the compiler to report the issue and not cover it up?

Comment (fortnern) [reply test/earray.c:2306]: I'm not certain the compiler will report an issue if there's no cast. And it is much more unsafe to pass the wrong type to printf than to cast something to the right type - the only potential problem there is wrap around. printf looks at the format string and reads that many bytes from the input parameters, it does  not check if enough data was passed in the parameters, so if the types don't match exactly it's possible for printf to read past the end of the input parameters which is a memory error.

Comment (fortnern) [reply test/earray.c:2306]: It looks like gcc does not warn by default but does if -Wformat is on

Comment (jhendersonHDF) [reply test/earray.c:2306]: I believe we have Wformat on by default for GCC and most of the other compilers. My main concern is that casting to int can hide other problems if the variable ever does change to something else, but it seems unlikely that it will at this point. Absolutely not an important issue though; either way is fine.

**Comment by byrnHDF** [comment]
This is a test not the library - requiring this scrutiny is beyond the usual for a test! Other tests have more issues then if a simple value (0,1,2,3) would ever be more then an int?!

**Review by jhendersonHDF — APPROVED** [review]

**Review by fortnern — APPROVED** [review]
